lib/msf/core/post/file.rb

```diff
@@ -692,4 +683,43 @@ def _unix_max_line_length
 
     line_max
   end
+
+
+  def stat(filename)
+     if session.type == 'meterpreter'
+      return session.fs.file.stat(filename)
+    end
+    if session.type == 'shell'
+      if session.platform == 'windows'
+        raise "`stat' method does not support Windows systems"
+      end
+      return FileStat.new(filename, session)
```

Comment:
This should check if the `stat` command exists to try and avoid an error.

Reply:
Okay, will add a check for that!

Reply:
Done :+1: